src/charm.py

```diff
@@ -78,6 +76,17 @@ def __init__(self, *args):
 
         self.restart = RollingOpsManager(self, relation="restart", callback=self._restart)
 
+        # --- COS ---
+        self.cos_integration = COSAgentProvider(
+            self,
+            relation_name=COS_RELATION_NAME,
+            metrics_endpoints=[],
+            scrape_configs=self._scrape_config,
+            refresh_events=[self.on.config_changed],
+            # metrics_rules_dir="./src/alert_rules/prometheus",
+            # log_slots=["opensearch:logs"],
```

Comment:
With the patch you brough on the snap, this slot can be re set now I think `opensearch-dashboards:log`

Reply:
Yes, I'm adding those changes all together once the snap version van be increased

Reply:
Done.